File: Python/0342PowerOfFour.py
def isPowerOfFour(n : int) -> bool:
    return n > 0 and (n & (n - 1)) == 0 and (n & 0x55555555) != 0

n = 5
print(isPowerOfFour(n))

n = 16
print(isPowerOfFour(n))

File: C/0342PowerOfFour.c
#include <stdio.h>
#include <stdbool.h>

bool isPowerOfFour(int n){
    return (n > 0) && ((n & (n - 1)) == 0) && ((n & 0x55555555) != 0);
}

int main(){

    int n = 16;
    printf("%d", isPowerOfFour(n));

    return 0;
}